asset/window/processor.js

```diff
@@ -6,12 +6,23 @@ const DataWindow = require('../__lib/data-window');
 class Window extends BatchProcessor {
     constructor(...args) {
         super(...args);
-        this.events = this.context.apis.foundation.getSystemEvents();
-        this.shuttingDown = false;
+        this.flushWindows = false;
         this.windows = new Map();
         this.results = [];
     }
 
+    shutdown() {
+        this.flushWindows = true;
```

Comment:
You might get issues when setting `this.flushWindows = true` here since a slice is usually flushed before shutdown and when it is not, it may not be safe to flush.

Reply:
I'm not really sure what to do on a shutdown.  Assuming there's a flush there is no need to set this.flushWindows = true on a shutdown.  But, if there's not a flush before a shutdown then maybe it doesn't matter and data will be lost no matter what.

Reply:
You don't really need to do anything in this case, if you let the framework ensure a flush is happening at the right, you are less likely to run into problems.